data/datasource.go

```diff
@@ -371,6 +390,15 @@ func readStdin(source *Source, args ...string) ([]byte, error) {
 	return b, nil
 }
 
+// readPlugin returns the plugin's get function
+// and throws an errors if the get functions does not exist
+func readPlugin(source *Source, args ...string) ([]byte, error) {
+	if source.gf == nil {
+		return nil, errors.Errorf("Datasource with %s is a plugin datasource but its plugin's get method is nil.", source.Alias)
+	}
+	return source.gf.(func(url *url.URL, headers http.Header, args []string) ([]byte, error))(source.URL, source.header, args)
```

Comment:
Also just notice: what if the plugin method doesn't have this format? will it error? we should do ```gf, ok := ource.gf.(func(url *url.URL, headers http.Header, args []string) ([]byte, error))``` first.